boot/zephyr/main.c

```diff
@@ -376,12 +426,50 @@ void main(void)
     __ASSERT(rc >= 0, "Error of the reading the detect pin.\n");
     if (detect_value == CONFIG_BOOT_SERIAL_DETECT_PIN_VAL &&
         !boot_skip_serial_recovery()) {
-        BOOT_LOG_INF("Enter the serial recovery mode");
-        rc = boot_console_init();
-        __ASSERT(rc == 0, "Error initializing boot console.\n");
-        boot_serial_start(&boot_funcs);
-        __ASSERT(0, "Bootloader serial process was terminated unexpectedly.\n");
-    }
+            
+        k_sleep(K_MSEC(50));
+
+        /* Get the uptime for debounce purposes. */
+        uint32_t timestamp = k_uptime_get();
+
+        for(;;) {
+            
+#ifdef GPIO_INPUT
+            rc = gpio_pin_get_raw(detect_port, CONFIG_BOOT_SERIAL_DETECT_PIN);
+            detect_value = rc;
+#else
+            rc = gpio_pin_read(detect_port, CONFIG_BOOT_SERIAL_DETECT_PIN,
+                            &detect_value);
+#endif
+            __ASSERT(rc >= 0, "Error of the reading the detect pin.\n");
+
+            uint32_t delta = k_uptime_get() -  timestamp;
+
+            /* If not pressed OR if pressed > debounce period stop loop*/
+            if( (detect_value == CONFIG_BOOT_SERIAL_DETECT_PIN_VAL && delta >= 
```

Comment:
if you really expected environmental noises, this check looks strange to me.

Reply:
I harsh environment medina value can be used as expected value.
In office environment signal should be stable after denounce time so even one sample might be used.

Reply:
How are we expecting this button to be used? I thought the button was to be held down before reset, in which case, I wouldn't expect any bounce.

Reply:
Another main use case for this button is that it's used to wake the device up from power down. Without this "debounce"  the device goes into Bootloader mode. It's an extra confirmation that the end user actually wants to go into the bootloaer.

Reply:
I would also agree we can make this optional. It's particular to the nRF9160 Feather series of boards. Not everyone cares about that extra startup time.

Reply:
Just pushed a fix for this. By default there is no boot delay. If there is one then this is added in.